ctpolicy/loglist/loglist.go

```diff
@@ -171,17 +174,24 @@ func (ll List) subset(names []string) (List, error) {
 }
 
 // forPurpose returns a new log list containing only those logs whose states are
-// acceptable for the given purpose. It returns an error if the purpose is
-// Issuance or Validation and the set of remaining logs is too small to satisfy
-// the Google "two operators" log policy.
-func (ll List) forPurpose(p purpose) (List, error) {
+// acceptable for the given purpose. Test logs are included only when
+// submitToTestLogs is true. It returns an error if the purpose is Issuance or
+// Validation and the set of remaining logs is too small to satisfy the Google
+// "two operators" log policy.
+func (ll List) forPurpose(p purpose, submitToTestLogs bool) (List, error) {
 	res := make(List, 0)
 	operators := make(map[string]struct{})
+
+	// Test logs in Chrome's all_logs_list.json omit the "state" field. loglist3
+	// interprets this as "UndefinedLogStatus", which causes usableForPurpose()
+	// to return false. To account for this, we skip this check for test logs.
```

Comment:
I feel like we've ended up in a weird abstraction layering situation here.

We're plumbing the test-ness of a log into forPurpose, because this method doesn't work correctly on test logs which don't have a log state. But the _reason_ it doesn't work correctly is that usuableForPurpose doesn't handle test logs well, and yet we're not modifying that helper function at all. Why does the understanding of test-ness stop at this particular layer of the stack?

Wouldn't it work just as well to pass log.Type into usableForPurpose?

Reply:
Good question. Personally, I don’t see this as an abstraction issue, if anything, I think usableForPurpose might not need to be a helper at all, and that's what feels weird about it.

It’s not reused, not externally exposed, and not separated for testability. If we tried to push this logic down to usableForPurpose, we’d need to pass both log.Type and submitToTestLogs, which just complicates things further. At that point, it might be simpler and clearer to inline the status check entirely.